### PR TITLE
gh-118943: Remove 'regen-jit' from the 'regen-all' make target

### PR DESCRIPTION
**Pull request by hroncok**
* Issue: gh-118943

### CODE REVIEW

**Review by Eclips4 — APPROVED** [review]

**Comment by erlend-aasland** [comment]
Workflow nit: instead of opening a PR against a closed issue, please either re-open the existing issue, or create a new one.

**Comment by erlend-aasland** [comment]
For the record, see https://github.com/python/cpython/pull/120690#pullrequestreview-2213830439 for why this PR was made.

**Review by erlend-aasland — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -85,7 +85,7 @@ jobs:
       - name: Build CPython
         run: |
           make -j4 regen-all
-          make regen-stdlib-module-names regen-sbom
+          make regen-stdlib-module-names regen-sbom regen-jit
```
Comment (erlend-aasland) [inline .github/workflows/build.yml]: I don't think this is needed (though I don't think it has any effect, since `regen-jit` is an empty target unless `--experimental-jit` is given).

**Review by erlend-aasland — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (erlend-aasland) [reply .github/workflows/build.yml]: FTR, see `.github/workflows/jit.yml` for the JIT CI.

File: Makefile.pre.in [hunk 2]
```diff
@@ -1665,10 +1665,10 @@ regen-unicodedata:
 regen-all: regen-cases regen-typeslots \
 	regen-token regen-ast regen-keyword regen-sre regen-frozen \
 	regen-pegen-metaparser regen-pegen regen-test-frozenmain \
-	regen-test-levenshtein regen-global-objects regen-jit
+	regen-test-levenshtein regen-global-objects
 	@echo
 	@echo "Note: make regen-stdlib-module-names, make regen-limited-abi, "
-	@echo "make regen-configure, make regen-sbom, and make regen-unicodedata should be run manually"
+	@echo "make regen-configure, make regen-sbom, regen-jit, and make regen-unicodedata should be run manually"
```
Comment (merwok) [inline Makefile.pre.in]:
```suggestion
	@echo "make regen-configure, make regen-sbom, make regen-jit, and make regen-unicodedata should be run manually"
```
to follow the existing style and be explicit

File: .github/workflows/build.yml (hunk 1 above)
Comment (hroncok) [reply .github/workflows/build.yml]: Good point

**Comment by hroncok** [comment]
I have made the requested changes; please review again

**Comment by hroncok** [comment]
The aarch64 CI failures seem unrelated to this PR.

File: .github/workflows/jit.yml [hunk 3]
```diff
@@ -32,9 +32,27 @@ jobs:
...
+          if test -n "$changes"; then
+            echo "Generated files not up to date."
+            echo "Perhaps you forgot to run make regen-jit. ;)"
+            echo "$changes"
+            echo ""
+            git diff --staged || true
+            exit 1
+          fi
+      - name: Build tier two interpreter
+        run: |
```
Comment (brandtbucher) [inline .github/workflows/jit.yml]: Can you revert the changes in this file? `jit_stencils.h` isn't checked in, and it's built automatically by the default `all` target. So there's no need to run `make regen-jit` manually or check whether it's "up-to-date".

File: Makefile.pre.in [hunk 4]
```diff
@@ -1665,10 +1665,10 @@ regen-unicodedata:
 regen-all: regen-cases regen-typeslots \
 	regen-token regen-ast regen-keyword regen-sre regen-frozen \
 	regen-pegen-metaparser regen-pegen regen-test-frozenmain \
-	regen-test-levenshtein regen-global-objects regen-jit
+	regen-test-levenshtein regen-global-objects
 	@echo
 	@echo "Note: make regen-stdlib-module-names, make regen-limited-abi, "
-	@echo "make regen-configure, make regen-sbom, and make regen-unicodedata should be run manually"
+	@echo "make regen-configure, make regen-sbom, make regen-jit, and make regen-unicodedata should be run manually"
```
Comment (brandtbucher) [inline Makefile.pre.in]: Please revert this too... nobody needs to `make regen-jit` manually like this.

File: Misc/NEWS.d/next/Build/2024-08-02-12-59-15.gh-issue-118943.vZQtET.rst [hunk 5]
```diff
@@ -0,0 +1,2 @@
+The ``regen-jit`` make target is no longer part of ``regen-all`` to avoid various
+race conditions when running ``regen-all``.
```
Comment (brandtbucher) [inline Misc/NEWS.d/next/Build/2024-08-02-12-59-15.gh-issue-118943.vZQtET.rst]: I'd rephrase this, since we still regenerate the JIT, just once instead of twice. Something like this:

```suggestion
Fix an issue where the experimental JIT could be built several times by the ``make regen-all`` target, leading to possible race conditions on heavily parallelized builds.
```

**Review by brandtbucher — CHANGES_REQUESTED** [review]
I think we literally just need to remove `regen-jit` from `regen-all`, nothing more. :)

**Review by brandtbucher — APPROVED** [review]
Thanks!

**Comment by hroncok** [comment]
I misunderstood the problem at first. I assumed calling  `make regen-jit` was still necessary.

However, I now see that the `jit_stencils.h` target calls the same command, which probably explains why it was called parallelly.
